stdlib/array.gr

```diff
@@ -386,6 +417,9 @@ provide let reduceRight = (fn, initial, array) => {
  * @param initial: The initial value to use for the accumulator on the first iteration
  * @param array: The array to iterate
  * @returns The final accumulator returned from `fn`
+ * 
+ * @example
+ * Array.reducei((a, b, i) => a + b + i, 100, [> 1, 2, 3, 4]) == 116
```

Comment:
Just a thought here really up to you I find building strings makes things a lot more clear on the order this runs like if you did something like 
```
Array.reducei((a, b, i) => a ++ b ++ toString(i), "A", [>"B", "C", "D"])
```
it just makes the order a bit more clear while still being a relativly simple example `reduce` is hard though and some confusion can be left either way this might be best added as a second example on this as well, I would wait for @phated to comment on thougths on this though as it is slightly more complex.